panic: 'reflect: call of reflect.Value.NumField...' at anonymous field

Panic with the following test code. 😇

```go
//panic: reflect: call of reflect.Value.NumField on int64 Value

type testStruct struct {
    time.Duration
}

func TestMergePanicCase(t *testing.T) {
    to := testStruct{}
    from := testStruct{}
    if err := Merge(&to, from); err != nil {
        t.Fail()
    }
}
```


```
--- FAIL: TestMergePanicCase (0.00s)
panic: reflect: call of reflect.Value.NumField on int64 Value [recovered]
        panic: reflect: call of reflect.Value.NumField on int64 Value

goroutine 38 [running]:
testing.tRunner.func1(0xc4200de000)
        /usr/local/go1.9.2/src/testing/testing.go:711 +0x2d2
panic(0x55f500, 0xc42000b1c0)
        /usr/local/go1.9.2/src/runtime/panic.go:491 +0x283
reflect.flag.mustBe(0x186, 0x19)
        /usr/local/go1.9.2/src/reflect/value.go:201 +0xc8
reflect.Value.NumField(0x5733c0, 0xc420014f18, 0x186, 0x5785a0)
        /usr/local/go1.9.2/src/reflect/value.go:1176 +0x34
github.com/imdario/mergo.hasExportedField(0x5733c0, 0xc420014f18, 0x186, 0x0)
        /home/yamasaki_masahide/go/src/github.com/imdario/mergo/merge.go:16 +0x5a
github.com/imdario/mergo.hasExportedField(0x5785a0, 0xc420014f18, 0x199, 0xc42004deb8)
        /home/yamasaki_masahide/go/src/github.com/imdario/mergo/merge.go:19 +0x18a
github.com/imdario/mergo.deepMerge(0x5785a0, 0xc420014f18, 0x199, 0x5785a0, 0x6773e0, 0x99, 0xc42008de40, 0x0, 0x677300, 0x99, ...)
        /home/yamasaki_masahide/go/src/github.com/imdario/mergo/merge.go:49 +0xf68
github.com/imdario/mergo.merge(0x573540, 0xc420014f18, 0x5785a0, 0x6773e0, 0x578500, 0x15fe8d01, 0xc420014f18)
        /home/yamasaki_masahide/go/src/github.com/imdario/mergo/merge.go:166 +0x250
github.com/imdario/mergo.Merge(0x573540, 0xc420014f18, 0x5785a0, 0x6773e0, 0x46e176, 0x5a223bab)
        /home/yamasaki_masahide/go/src/github.com/imdario/mergo/merge.go:146 +0x4e
github.com/imdario/mergo.TestMergePanicCase(0xc4200de000)
        /home/yamasaki_masahide/go/src/github.com/imdario/mergo/mergo_test.go:674 +0x78
testing.tRunner(0xc4200de000, 0x596f80)
        /usr/local/go1.9.2/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
        /usr/local/go1.9.2/src/testing/testing.go:789 +0x2de
FAIL    github.com/imdario/mergo        0.007s

shell returned 1
```

--- a/issue50_test.go
+++ b/issue50_test.go
@@ -1,0 +1,18 @@
+package mergo
+
+import (
+	"testing"
+	"time"
+)
+
+type testStruct struct {
+	time.Duration
+}
+
+func TestIssue50Merge(t *testing.T) {
+	to := testStruct{}
+	from := testStruct{}
+	if err := Merge(&to, from); err != nil {
+		t.Fail()
+	}
+}

--- a/merge.go
+++ b/merge.go
@@ -8,14 +8,12 @@
 
 package mergo
 
-import (
-	"reflect"
-)
+import "reflect"
 
 func hasExportedField(dst reflect.Value) (exported bool) {
 	for i, n := 0, dst.NumField(); i < n; i++ {
 		field := dst.Type().Field(i)
-		if field.Anonymous {
+		if field.Anonymous && dst.Field(i).Kind() == reflect.Struct {
 			exported = exported || hasExportedField(dst.Field(i))
 		} else {
 			exported = exported || len(field.PkgPath) == 0

